### PR TITLE
docs: annotate change in benchmark hardware

### PR DESCRIPTION
**Pull request by jbowens**
See cockroachdb/cockroach#117852.

### CODE REVIEW

**Review by RaduBerinde — APPROVED** [review]
:lgtm: 

Is there a way to have a "global" annotation for all benchmarks? I'd assume many would see changes.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/pebble/3290#-:-NqUVKffEouYBxhtcr0I:b-ohx19y)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @srosenberg)

File: docs/index.html [hunk 1]
```diff
@@ -70,6 +70,7 @@
         <div class="annotation" data-date="20230607">ycsb/F sampling bug</div>
         <div class="annotation" data-date="20230611">Switched to m6id.4xlarge
             (from 5d.4xlarge)</div>
+        <div class="annotation" data-date="20240209">Changed AWS machine type.</div>
```
Comment (srosenberg) [inline docs/index.html:73]: Should we add `#117852` for easy cross-referencing?

**Review by srosenberg — APPROVED** [review]

**Comment by srosenberg** [comment]
Linking roachperf annotation [PR](https://github.com/cockroachdb/roachperf/pull/161).

**Review by jbowens — COMMENTED** [review]
TFTRs!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/pebble/3290#-:-NqY67drD_6SLfhn76z6:bz5mair)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @srosenberg)

___
*[`docs/index.html` line 73 at r1](https://reviewable.io/reviews/cockroachdb/pebble/3290#gh-1486893826:-NqY66q0Csr86bYErEI_:b-896fix) ([raw file](https://github.com/cockroachdb/pebble/blob/9374e56e4d2118cd27b4fd288e579e7c04f91bd4/docs/index.html#L73)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

Should we add `#117852` for easy cross-referencing?
</blockquote></details>

Done.

**Pull request merged**